components/profile/playHistory.tsx

```diff
@@ -0,0 +1,237 @@
+    const newValue = e.target.value;
+
+    if (minDurationMinutes === intermediateMinDuration) {
+      return;
+    }
+
+    setIntermediateMinDuration(newValue);
+    setMinDurationMinutes(intermediateMinDuration);
+    setAccumulatedPlayHistory([]); // Clear the play history
+    setCursor(null); // Reset the cursor
+  };
+
+  const display = accumulatedPlayHistory && accumulatedPlayHistory.map((playAttempt, index) => {
+    const level = playAttempt.levelId as EnrichedLevel;
+    let durationInbetween = null;
+
+    if (prevEndTime) {
+      durationInbetween = moment.duration(playAttempt.startTime - prevEndTime, 'seconds').humanize();
+    }
+
+    prevEndTime = playAttempt.endTime;
+
+    const currentDate = moment.unix(playAttempt.startTime).format('MMMM Do, YYYY');
+    const showDate = currentDate !== prevDate;
+
+    prevDate = currentDate;
+
+    const isLeftAligned = index % 2 === 0;
+
+    return (
+      <div
+        key={playAttempt._id.toString()}
+        className='flex flex-col gap-2'
+        style={{ position: 'relative' }}
+      >
+        {showDate && (
+          <div className={'md:text-lg text-xs absolute top-50% transform -translate-y-1/2 text-gray-500 sm:left-[-10%] lg:left-[-50%]'}>
+
+            {currentDate}
+          </div>
+        )}
+
+        { durationInbetween && <span className='text-gray-500 dark:text-gray-400 justify-center self-center'>{durationInbetween} later</span> }
+        <div className={`flex flex-row items-center relative ${isLeftAligned ? 'flex-row-reverse' : ''}`}>
+          <div>
+            <div style={{
+              position: 'absolute',
+              width: '2px',
+              backgroundColor: '#ccc',
+              top: 0,
+              bottom: 0,
+              left: '50%',
+              transform: 'translate(-50%)',
+
+              zIndex: 0,
+            }} />
+            <div style={{
+              position: 'absolute',
+              height: '10px',
+              width: '10px',
+              backgroundColor: '#ccc',
+              borderRadius: '50%',
+              left: '50%',
+              top: '50%',
+              transform: 'translate(-50%, -50%)',
+              zIndex: 1
+            }} />
+          </div>
+          <div className={`${isLeftAligned ? '' : 'mr-10'}`}>
+            <SelectCard option={{
+              author: level.userId?.name,
+              height: Dimensions.OptionHeightLarge,
+              width: Dimensions.OptionWidth * (isMobile ? 0.8 : 1),
+              href: `/level/${level.slug}`,
+              id: playAttempt.levelId._id.toString() + '-' + playAttempt._id.toString(),
+              level: level,
+              stats: new SelectOptionStats(level.leastMoves, level.userMoves),
+              text: level.name,
+            }} />
+          </div>
+          <div className={'flex flex-col ' + (isLeftAligned ? 'mr-5 items-end text-right' : 'ml-5 items-start')}>
+            {moment.unix(playAttempt.startTime).format('h:mma')}
```

Comment:
This should match the local time zone (it's currently showing me EDT instead of PDT)